sale_product_set_layout/views/product_set.xml

```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+    <record id="view_product_set_form" model="ir.ui.view">
+        <field name="inherit_id" ref="sale_product_set.view_product_set_form" />
+        <field name="name">product.set.form</field>
+        <field name="model">product.set</field>
+        <field name="arch" type="xml">
+            <xpath
+                expr="//field[@name='set_line_ids']//field[@name='product_id']"
+                position="attributes"
+            >
+                <attribute
+                    name="attrs"
+                >{'required': [('display_type', '=', False)]}</attribute>
+            </xpath>
+            <xpath expr="//field[@name='set_line_ids']/tree" position="inside">
```

Comment:
I think the lines in this tree view shoul be editable, so more products/notes/sections can be added from outside the form view.